services/ext/api.go

```diff
@@ -610,6 +613,16 @@ func MakeMessagesRequestPayload(r MessagesRequest) ([]byte, error) {
 	return rlp.EncodeToBytes(payload)
 }
 
+func topicsToByteArray(topics []types.TopicType) [][]byte {
```

Comment:
Why not define this as a method for `MessagesRequest` in it's own file? This way it could be reused in the future.
Or is that a premature optimization thing?

Reply:
Probably early, these files very rarely changes, but worth doing if next time we re-use it